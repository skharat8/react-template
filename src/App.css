@font-face {
  font-family: "chocolateWithBrownie";
  src: url(./assets/fonts/chocolate-with-brownie.ttf);
}

h1 {
  font-family: chocolateWithBrownie;
  color: var(--secondary-color);
  text-align: center;
  margin-bottom: 0.3em;
}

.logo {
  width: 13rem;
  padding: 1.5em;
  will-change: filter;
  transition: filter 300ms;
}
.logo:hover {
  filter: drop-shadow(0 0 2em var(--secondary-color));
  transform: scale(1.1);
}
