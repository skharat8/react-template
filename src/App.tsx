import { useState } from "react";
import reactLogo from "./assets/react.svg";
import viteLogo from "./assets/vite.svg";
import "./App.css";

export default function App({ title }: AppProps) {
  const [count, setCount] = useState(0);
  // const [user, setUser] = useState<User | null>(null);

  return (
    <>
      <div>
        <a href="https://vitejs.dev" target="_blank" rel="noreferrer">
          <img src={viteLogo} className="logo" alt="Vite logo" />
        </a>
        <a href="https://react.dev" target="_blank" rel="noreferrer">
          <img src={reactLogo} className="logo react" alt="React logo" />
        </a>
      </div>
      <h1>{title}</h1>
      <div className="card">
        <button
          type="button"
          onClick={() => setCount(prevCount => prevCount + 1)}
        >
          count {count}
        </button>
        <p>
          Edit <code>src/App.tsx</code> and save to test HMR
        </p>
      </div>
      <p className="read-the-docs">
        Click on the Vite and React logos to learn more
      </p>
    </>
  );
}

type AppProps = {
  title?: string;
};

App.defaultProps = {
  title: "React + Vite",
};

// type ContextType = { user: User | null };
// Use with: <Outlet context={{ user } satisfies ContextType} />

// export function useUser() {
//   return useOutletContext<ContextType>();
// }
