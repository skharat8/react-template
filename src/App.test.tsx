import { ReactElement } from "react";
import { render, screen } from "@testing-library/react";
import { expect, test } from "vitest";
import userEvent from "@testing-library/user-event";
import App from "./App";

function setup(jsx: ReactElement) {
  // Use setup functions that compose together when you have a lot of tests
  // that do the same thing.
  return {
    user: userEvent.setup(),
    ...render(jsx),
  };
}

test("renders main page", () => {
  const { container } = setup(<App />);
  expect(container).toMatchSnapshot();
});

test("renders correct heading", () => {
  setup(<App title="Hello World" />);
  expect(screen.getByRole("heading").textContent).toMatch(/Hello World/i);
});

test("increments count after button click", async () => {
  const { user } = setup(<App />);
  const button = screen.getByRole("button", { name: "count 0" });

  await user.click(button);
  expect(button.textContent).toMatch(/^count 1$/i);
});
