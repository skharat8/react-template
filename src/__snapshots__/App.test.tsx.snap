// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders main page 1`] = `
<div>
  <div>
    <a
      href="https://vitejs.dev"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="Vite logo"
        class="logo"
        src="/public/vite.svg"
      />
    </a>
    <a
      href="https://react.dev"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="React logo"
        class="logo react"
        src="/src/assets/react.svg"
      />
    </a>
  </div>
  <h1>
    React + Vite
  </h1>
  <div
    class="card"
  >
    <button
      type="button"
    >
      count 
      0
    </button>
    <p>
      Edit 
      <code>
        src/App.tsx
      </code>
       and save to test HMR
    </p>
  </div>
  <p
    class="read-the-docs"
  >
    Click on the Vite and React logos to learn more
  </p>
</div>
`;
